### ISSUE TITLE
installation error

### ISSUE THREAD

**Issue by franklevis** [ridwanskaterock/cicool#356]
Every time I try to install the application, the url is changed from www.pippo.com to www.pippo.com:80 and everything stops.

![image](https://user-images.githubusercontent.com/55537569/74814303-3f435b00-52f7-11ea-9242-84e6dd05a8e4.png)

if I remove the door 80 then I load a language selection page but then it still freezes and it doesn't seem to have been loaded properly

![image](https://user-images.githubusercontent.com/55537569/74814379-6863eb80-52f7-11ea-93a2-304d5bf7e402.png)

thanks for any help

**Comment by franklevis** [comment]
I solved the first problem. The script MUST be installed in its "cicool" dir. Otherwise it doesn't work.

But now I have another error:

![image](https://user-images.githubusercontent.com/55537569/74829283-fef0d680-5310-11ea-92f2-5ac21cbfb4a6.png)

ity: 8192

Message: strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior

Filename: MX/Router.php

Line Number: 239

Backtrace:

File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
Line: 239
Function: strpos

File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
Line: 72
Function: set_class

File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
Line: 329
Function: _parse_routes

File: /web/htdocs/www.topdrives.it/home/cicool/index.php
Line: 339
Function: require_once

**Comment by franklevis** [comment]
Also fixed the second problem, the software is not compatible with the latest version of PHP (7.3). Version 7.2 is installed and everything is fine. Almost, because now I have a third problem. I hope I don't have to go on like this for the next few days.

Third problem:

When I first try to create a new CRUD the table is not loaded. As per attached image. I am asked to give SQL mode to the whole group. Clicking a link which then obviously gives an error because being a rental database I do not have root privileges. Nobody has them.

So, How do I solve this problem too?

![image](https://user-images.githubusercontent.com/55537569/74846564-f3ada300-5330-11ea-8663-cb66ab398e5d.png)

**Comment by franklevis** [comment]
Hello? Hello? Hello?
Is there anybody in there?
Just nod if you can hear me
Is there anyone at home?
Come on now

:)

**Comment by inglesuniversal** [comment]
Hello, I am facing the same problem. The lasted update from cicool is AWESOME, but I just happen to install a fresh NEW LAMP server running PHP 7.3 and ALL the rest of my backend apps are running much more smoothly... So, just like you... I BEG the author of the code to update the cicool script to meet PHP 7.3 standards and make those unwanted installation/wizard errors go away for GOOD!

Best regards @franklevis

**Comment by ridwanskaterock** [comment]
Hello sorry i am in the travel :)

For that can you follow intruction on the top page
Of does not work click link .....

You can try click link for it

Thanks

Pada tanggal Jum, 21 Feb 2020 08.27, Hugo Barbosa <notifications@github.com>
menulis:

**Comment by franklevis** [comment]
Maybe I don't understand.
If I click on the link it doesn't work. It gives me this mistake. Obviously I don't have root rights on mysql servers

A Database Error Occurred
Error Number: 1227

Access denied; you need (at least one of) the SUPER privilege(s) for this operation

set global sql_mode='STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';

Filename: /web/htdocs/www.topdrives.it/home/cicool/modules/web/controllers/Web.php

Line Number: 48

**Comment by ridwanskaterock** [comment]
Yes you need permission for access this function, you can contact web
administrator for it

Thank you

Pada tanggal Jum, 21 Feb 2020 15.10, Francesco <notifications@github.com>
menulis:

**Comment by franklevis** [comment]
In the description of the product it should be clearly written that to use it you need to have the root credentials on the MySql DB. Anyone who uses a paid service cannot have it for obvious reasons. We only have administrative privileges on our instance, certainly not on the whole DB. I don't understand the reason for this requirement. Other completely similar products do not have this requirement. Unusable product for me.

**Comment by agusecc** [comment]
> 
> 
> I solved the first problem. The script MUST be installed in its "cicool" dir. Otherwise it doesn't work.
> 
> But now I have another error:
> 
> ![image](https://user-images.githubusercontent.com/55537569/74829626-bbe33300-5311-11ea-83e2-47bba0cd0d05.png)
> 
> A PHP Error was encountered
> Severity: 8192
> 
> Message: strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior
> 
> Filename: MX/Router.php
> 
> Line Number: 239
> 
> Backtrace:
> 
> File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
> Line: 239
> Function: strpos
> 
> File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
> Line: 72
> Function: set_class
> 
> File: /web/htdocs/www.topdrives.it/home/cicool/application/third_party/MX/Router.php
> Line: 329
> Function: _parse_routes
> 
> File: /web/htdocs/www.topdrives.it/home/cicool/index.php
> Line: 339
> Function: require_once

**Comment by agusecc** [comment]
Please go to this link:
https://www.initekno.com/error-di-router-codeigniter-versi-3-1-10-dan-php-7/

**Comment by mangesh-sutar** [comment]
How can we integrate this Cicool tool with the existing Codeignitor project?
